### PR TITLE
Fix #94: Do not exit if box is not supported

### LINKED ISSUE #94: vagrant up for centos 7 box returns error if service-manager is installed
I have rebuild the service-manager from master (`commit 85ef24a93f6a9fd48dd62ca7730732e0d79456ec` ), installed it. Then did 
- vagrant init centos/7
- vagrant up  

``` r
$ vagrant up
Bringing machine 'default' up with 'libvirt' provider...
==> default: Creating image (snapshot of base box volume).
==> default: Creating domain with the following settings...
==> default:  -- Name:              centos_default
==> default:  -- Domain type:       kvm
==> default:  -- Cpus:              1
==> default:  -- Memory:            512M
==> default:  -- Management MAC:    
==> default:  -- Loader:            
==> default:  -- Base box:          centos/7
==> default:  -- Storage pool:      default
==> default:  -- Image:             /var/lib/libvirt/images/centos_default.img (41G)
==> default:  -- Volume Cache:      default
==> default:  -- Kernel:            
==> default:  -- Initrd:            
==> default:  -- Graphics Type:     vnc
==> default:  -- Graphics Port:     5900
==> default:  -- Graphics IP:       127.0.0.1
==> default:  -- Graphics Password: Not defined
==> default:  -- Video Type:        cirrus
==> default:  -- Video VRAM:        9216
==> default:  -- Keymap:            en-us
==> default:  -- INPUT:             type=mouse, bus=ps2
==> default:  -- Command line : 
==> default: Creating shared folders metadata...
==> default: Starting domain.
==> default: Waiting for domain to get an IP address...
==> default: Waiting for SSH to become available...
    default: 
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default: 
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if it's present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

grep VARIANT_ID /etc/os-release

Stdout from the command:



Stderr from the command:



```

```
[lmohanty@LalatenduM-laptop centos]$ vagrant ssh
[vagrant@localhost ~]$ cat /etc/os-release 
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

```

### PR DESCRIPTION
**Pull request by navidshaikh**
Fixes #94

### CODE REVIEW

File: plugins/guests/redhat/cap/osvariant.rb [hunk 1]
```diff
@@ -3,13 +3,12 @@ module GuestRedHat
     module Cap
       class OsVariant
         def self.os_variant(machine)
-          machine.communicate.sudo("grep VARIANT_ID /etc/os-release") do |type, data|
-            if type == :stderr
-              @env.ui.error(data)
-              exit 126
+          if machine.communicate.test("grep VARIANT_ID /etc/os-release")
+            machine.communicate.execute("grep VARIANT_ID /etc/os-release") do |type, data|
```
Comment (navidshaikh) [inline plugins/guests/redhat/cap/osvariant.rb]: I would like to make it more efficient by executing just one command inside the box than executing two.

Comment (LalatenduMohanty) [inline plugins/guests/redhat/cap/osvariant.rb]: @navidshaikh we can do 
if machine.communicate.test("grep CDK /etc/os-release")  return cdk
elseif 
machine.communicate.test("grep ADB /etc/os-release")  return ADB

Comment (LalatenduMohanty) [inline plugins/guests/redhat/cap/osvariant.rb]: else 
return nil

Comment (navidshaikh) [inline plugins/guests/redhat/cap/osvariant.rb]: This should do, however thinking out loud, for other boxes strings "CDK" or "ADB" could be a substring for other strings. Also this method does not check for ADB or CDK this just returns the VARIANT_ID value and check is happening somewhere else.

Comment (brgnepal) [inline plugins/guests/redhat/cap/osvariant.rb]: Please extract command `grep VARIANT_ID /etc/os-release` into variable.

File: plugins/guests/redhat/cap/osvariant.rb [hunk 2]
```diff
@@ -3,13 +3,14 @@ module GuestRedHat
...
-              exit 126
+          command = "grep VARIANT_ID /etc/os-release"
+          # TODO: execute efficient command to solve this
+          if machine.communicate.test(command) # test if command is exits with code 0
+            machine.communicate.execute(command) do |type, data|
+              return data.chomp.gsub(/"/, '').split("=").last
             end
-            return data.chomp.gsub(/"/, '').split("=").last
           end
+          return nil
```
Comment (brgnepal) [inline plugins/guests/redhat/cap/osvariant.rb]: Probably this line is not required. 
If `if` block doesn't execute then `nil` will be returned by default. Please cross-check and update it.

**Comment by navidshaikh** [comment]
@bexelbie @budhrg please review

**Comment by LalatenduMohanty** [comment]
@navidshaikh LGTM

**Comment by brgnepal** [comment]
@navidshaikh LGTM :+1:

**Pull request merged**